testsuite/pytests/test_stdp_synapse.py

```diff
@@ -87,15 +89,51 @@ def do_nest_simulation_and_compare_to_reproduced_weight(self, fname_snip):
                 title_snip=self.nest_neuron_model + " (NEST)",
             )
 
-        t_weight_reproduced_independently, weight_reproduced_independently = self.reproduce_weight_drift(
-            pre_spikes, post_spikes, self.init_weight, fname_snip=fname_snip
-        )
+        (
+            t_weight_reproduced_independently,
+            weight_reproduced_independently,
+            Kpre_log,
+            Kpost_log,
+        ) = self.reproduce_weight_drift(pre_spikes, post_spikes, self.init_weight, fname_snip=fname_snip)
 
         # ``weight_by_nest`` contains only weight values at pre spike times, ``weight_reproduced_independently``
-        # contains the weight at pre *and* post times: check that weights are equal for pre spike times
+        # contains the weight at pre *and* post times: check that weights are equal only for pre spike times
         assert len(weight_by_nest) > 0
-        np.testing.assert_allclose(t_weight_by_nest, t_weight_reproduced_independently)
-        np.testing.assert_allclose(weight_by_nest, weight_reproduced_independently)
+
+        difference_matrix = t_weight_by_nest[t_weight_by_nest < self.simulation_duration].reshape(
```

Comment:
Why this more complicated formulation of the assert on times?

Reply:
Weight recorders in master only record pre-synaptic spikes and this will change in the future. As a preparation for the future, the weights at all times are calculated and returned in the test. The complicated asserts then extract the weights changes due to pre-synaptic spikes from all these weights. Not sure if there was any other reason, the test is already a year old, I can't quite remember all details.